packages/js/components/src/utils.tsx

```diff
@@ -77,10 +82,18 @@ export { createOrderedChildren };
  * @param {Array} fills - slot's `Fill`s.
  * @return {Node} Node.
  */
-export const sortFillsByOrder: Slot.Props[ 'children' ] = ( fills ) => {
+export const sortFillsByOrder: SlotProps[ 'children' ] = ( fills ) => {
 	// Copy fills array here because its type is readonly array that doesn't have .sort method in Typescript definition.
-	const sortedFills = [ ...fills ].sort( ( a, b ) => {
-		return a[ 0 ].props.order - b[ 0 ].props.order;
+	const sortedFills = Children.toArray( fills ).sort( ( a, b ) => {
+		if (
+			typeof a === 'object' &&
+			'key' in a &&
+			typeof b === 'object' &&
+			'key' in b
+		) {
+			return a.props.order - b.props.order;
```

Comment:
Reminder to myself: To test this and write some testing instructions for this change.
As this did change from `a[0]` to `a.props`, which in theory should break this functionality, but the typing seems to disagree.

Reply:
I updated the test instructions to include this case.
Turns out that the `Children.toArray` actually flattens the fills for us and allows us to get access to the `props` directly.